src/base/rss/rssfolder.cpp

```diff
@@ -189,7 +189,6 @@ QHash<QString, FeedPtr> Folder::getAllFeedsAsHash() const
 
 bool Folder::addFile(const FilePtr &item)
 {
-    Q_ASSERT(!m_children.contains(item->id()));
```

Comment:
If you call this with an existing item, the if statement right below it will prevent it from being added, resulting in the function call being a no-op. IMO, that is the correct behavior, and what I would expect to happen. The fact that it causes a crash in debug mode I think is pointless. If you add an item to a Set that already has the item, it doesn't cause the program to crash, it just silently ignores it. It's the same idea here.